### DISCUSSION TITLE
What do you see missing in tts project and how do you think we can improve it

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#67] [Archived]
<USER>>>> erogol<br/> <DATE> [July 17, 2019, 1:47pm]<br/> <br/> Please here share your comments, opinions about TTS and let us discuss<br/> together to enable a better open-sourced environment.<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/what-do-you-see-missing-in-tts-project-and-how-do-you-think-we-can-improve-it]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> geneing<br/> <DATE> [July 18, 2019, 6:24am]<br/> <br/> active development.<br/> done and current status.<br/> to checkout old git versions to use pre-trained weights. It may be<br/> hard to do with PyTorch.<br/> quantization.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [July 18, 2019, 12:24pm]<br/> <br/> ![](https://cdn-production.discourse.mozilla-community.org/letter_avatar_proxy/v4/letter/g/bbce88/40.png){.avatar<br/> <br/> > - Better communication about the roadmap and what features are in<br/> > active development.<br/> > - Better information about different active branches - what's being<br/> > done and current status.<br/> ><br/> }<br/> > <br/> > ```<br/> > - This is true and needed to be fixed.<br/> ><br/> }<br/> > <!-- --><br/> > ```<br/> > - Backward compatibility with pre-trained models. Currently, we need<br/> > to checkout old git versions to use pre-trained weights. It may be<br/> > hard to do with PyTorch.<br/> > - Better integration with WaveRNN. Maybe include it as a subproject.<br/> > - Some work on improving deployment, e.g. weight pruning and<br/> > quantization.<br/> <br/> we need to be more clear.<br/> <br/> development. We should have a more stable code-base to promise this.<br/> <br/> fully included in TTS increases the complexity which can not be<br/> taken with the current head count of the project. If anyone is<br/> volunteer on this, that's always welcome.<br/> <br/> contributions are always welcome. However, before worrying about the<br/> deployment, I like to fill any algorithmic gap in TTS since any<br/> change in the algorithmic level will directly effect the deployment<br/> phase.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> geneing<br/> <DATE> [July 21, 2019, 5:49pm]<br/> <br/> It would be nice to have a place to discuss different experiments with<br/> improving the network architecture. If we could agree on a standard<br/> training/test set and on a standard vocoder, maybe we could compare<br/> results of experiments done by different people. Without standard<br/> procedure, it's no possible to tell if the improvement is due to changes<br/> to the architecture or due to a better training set.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> alchemi5t<br/> <DATE> [July 22, 2019, 6:16am]<br/> <br/> Very inconsistent between experiments. Might want to check performance<br/> vs the tensorflow variants. Keithito's implementation is quite<br/> consistent between various datasets.<br/> <br/> I am pro pytorch, but it's kinda hard to continue development here when<br/> results are this inconsistent. I know I haven't given enough information<br/> for you guys to work on, just wanted to make sure you guys compare<br/> performance vs the tensorflow variants.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [July 22, 2019, 2:24pm]<br/> <br/> We can easily discuss experiments on Github issues. If you like to run a<br/> new one, you can just create a new issue and we go for it. I try to keep<br/> my experiments up-to-date on Github as explained. Let me know, if it is<br/> not enough or you have a better way.<br/> <br/> For the vocoder, TTS's default vocoder is GL so it is always easier to<br/> experiment using it. All the released models are also with GL vocoder,<br/> so if anyone likes to compare results, he can just use the latest<br/> release for the target architecture and compare with his version. I<br/> guess, it is better to share tensorboard logs as well, if you like to<br/> compare loss curves. Does it make sense?<br/> <br/> All releases, except one is with LJSpeech, so all benchmarking is going<br/> around it.<br/> <br/> Let me know, if you have a better way to follow collaborative<br/> experimentation.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [July 22, 2019, 2:25pm]<br/> <br/> what you mean by inconsistent results?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> alchemi5t<br/> <DATE> [July 22, 2019, 3:37pm]<br/> <br/> That sounds pretty good! But I think it'll be better if we discuss<br/> experiments on discourse and leave bug fixes for github issues. GL<br/> vocoder is perfectly fine, that's what is used on the tensorflow<br/> implementations as well.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [July 22, 2019, 3:41pm]<br/> <br/> I just don't like to force people to have discourse account. We can tag<br/> the issues as 'experiment', then we can easily filter them.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> alchemi5t<br/> <DATE> [July 22, 2019, 3:44pm]<br/> <br/> By inconsistent I mean, when I use similar datasets to train the tts on<br/> the tensorflow variant(I'll link the repo soon) I dont have to tune any<br/> hyperparameters at all there(I.E., I tune it to work well on a standard<br/> dataset and once I hit the sweet spot, I train another model on a<br/> different dataset with a similar composition including quality(sample<br/> rate and a few other params) and that works out wonderfully). But on the<br/> mozilla one, I find that, once I tune the hyper parameter to the same<br/> standard dataset and then when I try to train a model on another<br/> dataset, it takes ages(1-2 days) to get past just producing noise and<br/> then it starts sounding like the person's voice but the content is just<br/> gibberish. I am not currently invested into experimenting with the TTS<br/> system, so I won't be able to give you concrete information to go by.<br/> I'll be back in a week or two and I'll try and explain the situation<br/> with a lot more concrete information!<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> alchemi5t<br/> <DATE> [July 23, 2019, 3:19am]<br/> <br/> ![](https://cdn-production.discourse.mozilla-community.org/user_avatar/discourse.mozilla.org/erogol/40/27780_2.png){.avatar<br/> <br/> > I just don't like to force people to have discourse account. We can<br/> > tag the issues as 'experiment', then we can easily filter them.<br/> <br/> Oh that's cool too. But since I have one already, might as well build a<br/> community here! I did believe having a discourse account was a pain, but<br/> I've realized helping build a community is actually quite rewarding!<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> josh_meyer<br/> <DATE> [July 25, 2019, 6:38pm]<br/> <br/> +1 on the WaveRNN integration<br/> <br/> [Archived Post]<br/>
